rfc/0054-sdk3-management-apis.md

```diff
@@ -342,12 +342,29 @@ Iterable<QueryIndex> GetAllIndexes(string bucketName, [options])
 * Optional:
 
   * `Timeout` or `timeoutMillis` (`int`/`duration`) - the time allowed for the operation to be terminated. This is controlled by the client.
+  * `CollectionName` - the name of the collection to restrict indexes to.
+  * `ScopeName` - the name of the scope to restrict indexes to.
 
 ### N1QL
 
+No collections set, returns all indexes for the given bucket, for all scopes and collections:
+```
+SELECT idx.* FROM system:indexes AS idx
+WHERE ((bucket_id IS MISSING AND keyspace_id = "bucketName") OR bucket_id = "bucketName") AND `using`="gsi"
+ORDER BY is_primary DESC, name ASC
+```
+
+Collection and scope set, returns all indexes for the given collection in the given scope, in the given bucket:
+```
+SELECT idx.* FROM system:indexes AS idx
+WHERE keyspace_id = "collectionName" AND bucket_id= "bucketName" AND scope_id = "scopeName" AND `using`="gsi"
```

Comment:
Should "Collection and scope set" be split into two cases, depending on whether the collection is the default collection? Otherwise, the client never finds indexes in the default collection. Suggestion:

----

Collection and scope set, and CollectionName is "_default". Returns all indexes for the default collection in the given scope, in the given bucket.
```
SELECT idx.* FROM system:indexes AS idx
WHERE ((bucket_id = "bucketName" AND scope_id = "scopeName" AND keyspace_id = "collectionName") OR (bucket_id IS MISSING AND keyspace_id = "bucketName")) AND `using`="gsi"
ORDER BY is_primary DESC, name ASC
``` 

Collection and scope set, and CollectionName is NOT "_default". Returns all indexes for the given collection in the given scope, in the given bucket.
```
SELECT idx.* FROM system:indexes AS idx
WHERE (bucket_id = "bucketName" AND scope_id = "scopeName" AND keyspace_id = "collectionName") AND `using`="gsi"
ORDER BY is_primary DESC, name ASC
```

Reply:
If the user sets the collection name to "_default" then I think that the existing query would work. 
```
((bucket_id = "bucketName" AND scope_id = "scopeName" AND keyspace_id = "collectionName") OR (bucket_id IS MISSING AND keyspace_id = "bucketName"))
```
I think that the latter part of this OR would result in all indexes being returned not only the ones in the _default collection?

Reply:
> I think that the latter part of this OR would result in all indexes being returned not only the ones in the _default collection?

Are you referring to `OR (bucket_id IS MISSING AND keyspace_id = "bucketName")` ? This only matches indexes on the default collection, because an index on a non-default collection always has a `bucket_id`.

> If the user sets the collection name to "_default" then I think that the existing query would work.

Please can you try it and confirm? The reason I expect it to fail is that the JSON for an index on the default collection index always looks like this, regardless of the Couchbase Server version:

```json
[
  {
    "datastore_id": "http://127.0.0.1:8091",
    "id": "1fd61fd5f9e1a0d4",
    "index_key": [
      "`foo`",
      "`bar`"
    ],
    "keyspace_id": "my-bucket",
    "name": "mytest",
    "namespace_id": "default",
    "state": "deferred",
    "using": "gsi"
  }
]
```

A query that looks for `bucket_id = "bucketName"` will never see a default index.